vignettes/faq-reordering.Rmd

```diff
@@ -0,0 +1,231 @@
+---
+title: "FAQ: Reordering"
+output: rmarkdown::html_vignette
+description: |
+  Changing the order in which data are plotted or presented on a plot.
+vignette: >
+  %\VignetteIndexEntry{FAQ: Reordering}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{=html}
+<style>
+  .content h3 {
+    margin-top: -30px !important;
+  }
+  
+  details {
+    margin-bottom: 40px;
+  }
+</style>
+```
+```{r, include = FALSE}
+library(ggplot2)
+library(dplyr)
+library(tibble)
+
+knitr::opts_chunk$set(
+  fig.dpi = 300, 
+  collapse = TRUE, 
+  comment = "#>",
+  fig.asp = 0.618,
+  fig.width = 6,
+  out.width = "80%"
+  )
+```
+
+## Bar plots
+
+### How can I reorder the bars in a bar plot by their value?
+
+Change the order of the levels of the factor variable you're creating the bar plot for in the `aes`thetic `mapping`.
+The forcats package offers a variety of options for doing this, such as `forcats::fct_reorder()`.
+
+<details>
+
+<summary>See example</summary>
+
+The following bar plot shows the number of cars that fall into each `class` category.
+Classes are ordered alphabetically.
+You might prefer them to be ordered by the number of cars in each class.
+
+```{r}
+ggplot(mpg, aes(y = class)) +
+  geom_bar()
+```
+
+To do this, you should first calculate the number of cars that fall into each `class` category.
```

Comment:
There's various ways to avoid manually calculating the sums, and in my experience (teaching a class on this right now) students constantly confuse `geom_bar()` and `geom_col()` so I would try to avoid switching right away.

```r
ggplot(mpg, aes(y = fct_infreq(class))) +
  geom_bar()

ggplot(mpg, aes(y = fct_rev(fct_infreq(class)))) +
  geom_bar()

ggplot(mpg, aes(y = fct_reorder(class, class, length))) +
  geom_bar()
```

Reply:
Oh interesting. I've had the opposite experience, but might be a function of teaching dplyr and ggplot2 together (already doing lots of `count()` so they expect to use it). I'll update to `fct_infreq()` since it's basically designed for that, I think the cognitive load on `fct_reorder(class, class, length)` is even higher.